avocado/core/job.py

```diff
@@ -505,11 +505,14 @@ def _run(self):
             archive.create(filename, self.logdir)
         _TEST_LOGGER.info('Test results available in %s', self.logdir)
 
-        tests_status = not bool(failures)
-        if tests_status:
-            return exit_codes.AVOCADO_ALL_OK
-        else:
+        if 'INTERRUPTED' in summary:
+            return exit_codes.AVOCADO_JOB_INTERRUPTED
+        elif summary:
             return exit_codes.AVOCADO_TESTS_FAIL
+        elif summary is None:
+            return exit_codes.AVOCADO_JOB_FAIL
+        else:
+            return exit_codes.AVOCADO_ALL_OK
```

Comment:
Actually not. First you need to check if `summary is None`, otherwise you get an exception.